src/main/java/dev/Roach/GameLookup.java

```diff
@@ -38,10 +38,15 @@ public GameDealResponse giveTitleToGetListOFDealsWithStores(String gameTitle) {
         }
 
         if (!foundGame) {
-            System.out.println("No matches found for the title: " + gameTitle);
-            return new GameDealResponse();
+            throw new NoSuchElementException("No game deals found for the title: " + gameTitle);
         }
 
-        return gamesFetcher.getGameDealObjectUsingID(gameIdentificator);
+        GameDealResponse response = gamesFetcher.getGameDealObjectUsingID(gameIdentificator);
+
+        if (response.isEmpty()) {
+            throw new NoSuchElementException("GameDealResponse is empty for game ID: " + gameIdentificator);
```

Comment:
Please make the message more user friendly, users will have no clue what GameDealResponse is